storage/src/main/java/com/tencent/rss/storage/handler/impl/MemoryClientReadHandler.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.storage.handler.impl;
+
+import com.google.common.collect.Lists;
+import com.tencent.rss.client.api.ShuffleServerClient;
+import com.tencent.rss.client.request.RssGetInMemoryShuffleDataRequest;
+import com.tencent.rss.client.response.RssGetInMemoryShuffleDataResponse;
+import com.tencent.rss.common.BufferSegment;
+import com.tencent.rss.common.ShuffleDataResult;
+import com.tencent.rss.common.exception.RssException;
+import com.tencent.rss.common.util.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class MemoryClientReadHandler extends AbstractFileClientReadHandler {
```

Comment:
IMHO it may not be appropriate to have a sepecific *Memory* handler here, the client read data either from server or remote fs directly and it should not know storage deatils in the server side when read data from the server.